applications/callflow/src/module/cf_voicemail.erl

```diff
@@ -1390,18 +1398,26 @@ get_caller_id_number(Call) ->
 
 -spec maybe_save_meta(pos_integer(), mailbox(), whapps_call:call(), ne_binary(), wh_json:object()) ->
                              'ok'.
-maybe_save_meta(Length, #mailbox{delete_after_notify='false'}=Box, Call, MediaId, _UpdateJObj) ->
+maybe_save_meta(Length, #mailbox{after_notify_action='nothing'}=Box, Call, MediaId, _UpdateJObj) ->
     save_meta(Length, Box, Call, MediaId);
-maybe_save_meta(Length, #mailbox{delete_after_notify='true'}=Box, Call, MediaId, UpdateJObj) ->
+
+maybe_save_meta(Length, #mailbox{after_notify_action=Action}=Box, Call, MediaId, UpdateJObj) ->
     case wh_json:get_value(<<"Status">>, UpdateJObj) of
         <<"completed">> ->
-            {'ok', _} = couch_mgr:del_doc(whapps_call:account_db(Call), MediaId),
-            lager:debug("attachment was sent out via notification, deleted media file");
+            case Action of
+                'delete' ->
+                    {'ok', _} = couch_mgr:del_doc(whapps_call:account_db(Call), MediaId),
+                    lager:debug("attachment was sent out via notification, deleted media file");
+                'save' ->
+                    update_folder(?FOLDER_SAVED, MediaId, Box, Call),
+                    lager:debug("attachment was sent out via notification, saved media file")
+            end;
```

Comment:
I'd personally put the log lines before the code, using ING instead of ED. So that log-reading people get an idea of what was going on.
What do you think?